pkg/task/registration.go

```diff
@@ -0,0 +1,134 @@
+package task
+
+import (
+	"sync"
+
+	"strings"
+
+	"github.com/Peripli/service-broker-proxy/pkg/platform"
+	"github.com/Peripli/service-broker-proxy/pkg/sm"
+	"github.com/sirupsen/logrus"
+)
+
+const ProxyBrokerPrefix = "sm-proxy-"
+
+type SBProxyRegistration struct {
+	group          *sync.WaitGroup
+	platformClient platform.Client
+	smClient       sm.Client
+	smHost         string
+}
+
+type serviceBrokerReg struct {
+	*platform.ServiceBroker
+	SmID string
+}
+
+func New(group *sync.WaitGroup, platformClient platform.Client, smClient sm.Client, smHost string) *SBProxyRegistration {
+	return &SBProxyRegistration{
+		group:          group,
+		platformClient: platformClient,
+		smClient:       smClient,
+		smHost:         smHost,
+	}
+}
+
+func (r SBProxyRegistration) Run() {
+	r.group.Add(1)
+	defer r.group.Done()
+	r.run()
+}
+
+func (r SBProxyRegistration) run() {
+	logrus.Debug("Running broker registration task...")
+	registeredBrokers, err := r.platformClient.GetBrokers()
+	if err != nil {
+		logrus.Error("An error occurred while obtaining already registered brokers: ", err)
+		return
+	}
+
+	brokersFromPlatform := make([]serviceBrokerReg, len(registeredBrokers))
+	for _, broker := range registeredBrokers {
+		if !r.isBrokerProxy(broker) {
+			logrus.WithFields(logFields(&broker)).Debug("Registration task SKIPPING registered broker as is not recognized to be proxy broker...")
+			continue
+		}
+		brokerReg := serviceBrokerReg{
+			ServiceBroker: &broker,
```

Comment:
this needs to be a literal, as the reference changes in each iteration of the loop